src/grep.js

```diff
@@ -2,16 +2,55 @@ var common = require('./common');
 var fs = require('fs');
 
 common.register('grep', _grep, {
-  globStart: 2, // don't glob-expand the regex
+  globStart: _getGlobStart, // don't glob-expand the regex
   canReceivePipe: true,
-  cmdOptions: {
-    'v': 'inverse',
-    'l': 'nameOnly',
-    'i': 'ignoreCase',
-    'n': 'lineNumber',
-  },
 });
 
+// Gets the index of the first argument after `regex_filter`
+function _getGlobStart(args) {
+  return args[0] === '--'
+    ? 2
+    : Array.prototype.findIndex.call(args, function (v) {
+        return (
+          v instanceof RegExp ||
+          (typeof v === 'string' && v !== '' && !v.startsWith('-'))
+        );
+      }) + 1;
+}
+
+// Gets the value of the `beforeContext` or `afterContext` options, removing
+// them from the `options` string
+function _getContextOption(options, option) {
+  var regex = new RegExp('-(\\w*)' + option + '([A-Za-z]*) ?(\\d*)(.*)');
+  var value = null;
+  var replaced = options.replace(regex, function (_, before, after, val, rest) {
+    var result = '';
+    if (before || after || rest) {
+      result = (before + after + rest).trim();
+      if (result && !result.startsWith('-')) {
+        result = '-' + result;
+      }
+    }
+    value = parseInt(val);
```

Comment:
Please specify `parseInt(val, 10)`. Then you can remove the `radix` eslint suppression.

Reply:
We have some code in shelljs which actually does parseInt using a different radix (see chmod.js and exec.js), so I would like all calls to be explicit about the radix.

Reply:
This line is no longer needed.